acceptance/tests/facts/osx_numeric_hostname.rb

```diff
@@ -0,0 +1,23 @@
+test_name 'Querying the fqdn with a numeric hostname should not fail' do
+
+  # calling getaddrinfo with a numeric value on OS X does not fill the
+  # ai_canonname field of the addrinfo structure
+  confine :to, :platform => /^osx-/
+
+  agents.each do |agent|
+    original_hostname = agent.hostname.split('.').first
```

Comment:
`agent.hostname` is only hostname or full fqdn?

Reply:
Thought it was hostname at first, but after testing I found out that it's actually a fqdn.